ui/lib/service-worker-authenticated-download/service-worker-registration/index.js

```diff
@@ -1,32 +1,6 @@
 import { addSuccessHandler } from 'ember-service-worker/service-worker-registration';
-import Namespace from '@ember/application/namespace';
-
-function getToken() {
-  // fix this later by allowing registration somewhere in the app lifecycle were we can have access to
-  // services, etc.
-  return Namespace.NAMESPACES_BY_ID['vault'].__container__.lookup('service:auth').currentToken;
-}
 
 addSuccessHandler(function (registration) {
-  // attach the handler for the message event so we can send over the auth token
```

Comment:
So I _think_ the only issue here is that the handler in `init` won't be called on the initial installation of the worker. This handler gets called when the worker is registered: https://github.com/DockYard/ember-service-worker/blob/4564e495666865b80fd29daf15784e2801b59d1b/service-worker-registration/index.js#L13 - so the check in the `init` function will only be truthy on subsequent tries.

Reply:
I was wondering about that too. I was thinking since the event listener for the message is added when the component initializes then it would be listening before the user has a chance to click the download action :thinking:.

Reply:
OK I think you're right - I looked at the file I linked again (lol), and it runs when the JS package loads, so it should be installed by the time you get to the download page (or by the time it renders). Sorry it's been a little bit since I looked at this 😅 !

Reply:
Great thanks for talking through it with me! This is the first time I have looked at that part of the code and wanted to make sure I am understanding things.